tests/Silverback.Integration.Tests.E2E/Kafka/ErrorHandlingTests.cs

```diff
@@ -1177,5 +1265,101 @@ public async Task MovePolicy_ToOtherTopicAfterRetry_MessageRetriedAndMoved()
+                                            policy => policy.MoveToKafkaTopic(
+                                                moveEndpoint => moveEndpoint.ProduceTo("other-topic")))
+                                        .Configure(
+                                            config =>
+                                            {
+                                                config.GroupId = "consumer1";
+                                            })))
+                        .AddIntegrationSpy()
+                        .AddDelegateSubscriber(
+                            (IIntegrationEvent _) => throw new InvalidOperationException("Move!")))
+                .Run();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+            await publisher.PublishAsync(message1);
+            await publisher.PublishAsync(message2);
+            await publisher.PublishAsync(message3);
+            await Helper.WaitUntilAllMessagesAreConsumedAsync();
+
+            Helper.Spy.OutboundEnvelopes.Should().HaveCount(6);
+            Helper.Spy.InboundEnvelopes.Should().HaveCount(3);
+
+            var otherTopic = Helper.GetTopic("other-topic");
+            var otherTopicMessages = otherTopic.GetAllMessages();
+            otherTopic.MessagesCount.Should().Be(3);
+            otherTopicMessages[0].Value.Should()
+                .BeEquivalentTo(Helper.Spy.InboundEnvelopes[0].RawMessage.ReReadAll());
+            otherTopicMessages[1].Value.Should()
+                .BeEquivalentTo(Helper.Spy.InboundEnvelopes[1].RawMessage.ReReadAll());
+            otherTopicMessages[2].Value.Should()
+                .BeEquivalentTo(Helper.Spy.InboundEnvelopes[2].RawMessage.ReReadAll());
+
+            Helper.Spy.OutboundEnvelopes[5].Endpoint.Name.Should().Be("other-topic");
+            Helper.Spy.OutboundEnvelopes[5].Headers
+                .Should().ContainEquivalentOf(
+                    new MessageHeader(
+                        DefaultMessageHeaders.FailedAttempts,
+                        1));
+            Helper.Spy.OutboundEnvelopes[5].Headers
+                .Should().ContainEquivalentOf(
+                    new MessageHeader(
+                        DefaultMessageHeaders.FailureReason,
+                        "System.Reflection.TargetInvocationException in System.Private.CoreLib"));
+            Helper.Spy.OutboundEnvelopes[5].Headers
+                .Count(header => header.Name == DefaultMessageHeaders.SourceTimestamp)
+                .Should().Be(1);
+            Helper.Spy.OutboundEnvelopes[5].Headers
+                .Any(header => header.Name == DefaultMessageHeaders.TimestampKey)
+                .Should().BeFalse();
+            Helper.Spy.OutboundEnvelopes[5].Headers
+                .Should().ContainEquivalentOf(
+                    new MessageHeader(
+                        KafkaMessageHeaders.SourcePartition,
+                        "default-e2e-topic[0]"));
```

Comment:
Are you sure we don't just expect the partitionm index in this header? (Since we have a separate one for topic.)

Reply:
Fixed, thanks!